File: javascript/arrays/left_rotation.js
rotLeft = (a, d) =>{
    let b = [...a];
    
    [...Array(a.length).keys()].forEach(i=>{
        let index = (i + d) % a.length;
        b[i] = a[index];
    });

    return b;
};

console.log(rotLeft([1, 2, 3, 4, 5,], 4));

File: python/arrays/left_rotation.py
def rotLeft(a, d):

    b = a.copy()
    for i in range(0, len(a)):
        index = (i + d) % len(a)
        b[i] = a[index]
    
    return b

print(rotLeft([1, 2, 3, 4, 5,], 4))